packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-field-auth-iam.ts

```diff
@@ -0,0 +1,634 @@
+      await setupUser(userPoolId, cognito_username, cognito_password);
+      await signInUser(cognito_username, cognito_password);
+      const authCredentials = await Auth.currentCredentials();
+
+      const unauthAppSyncClient = getConfiguredAppsyncClientIAMAuth(apiEndPoint, appRegion, unAuthCredentials);
+      const authAppSyncClient = getConfiguredAppsyncClientIAMAuth(apiEndPoint, appRegion, authCredentials);
+      person1IAMPrivateClient = constructModelHelper('Person1', authAppSyncClient);
+      person2IAMPrivateClient = constructModelHelper('Person2', authAppSyncClient);
+      person3IAMPrivateClient = constructModelHelper('Person3', authAppSyncClient);
+      person1IAMPublicClient = constructModelHelper('Person1', unauthAppSyncClient);
+      person2IAMPublicClient = constructModelHelper('Person2', unauthAppSyncClient);
+      person3IAMPublicClient = constructModelHelper('Person3', unauthAppSyncClient);
+    };
+
+    afterAll(async () => {
+      const metaFilePath = path.join(projRoot, 'amplify', '#current-cloud-backend', 'amplify-meta.json');
+      if (existsSync(metaFilePath)) {
+        await deleteProject(projRoot);
+      }
+      deleteProjectDir(projRoot);
+      await cleanupDatabase();
+    });
+
+    const setupDatabase = async (): Promise<void> => {
+      const dbConfig = {
+        identifier,
+        engine,
+        dbname: database,
+        username,
+        password,
+        region,
+      };
+
+      const db = await setupRDSInstanceAndData(dbConfig, queries);
+      port = db.port;
+      host = db.endpoint;
+    };
+
+    const cleanupDatabase = async (): Promise<void> => {
+      await deleteDBInstance(identifier, region);
+    };
+
+    const initProjectAndImportSchema = async (): Promise<void> => {
+      await initJSProjectWithProfile(projRoot, {
+        disableAmplifyAppCreation: false,
+        name: projName,
+      });
+
+      const metaAfterInit = getProjectMeta(projRoot);
+      region = metaAfterInit.providers.awscloudformation.Region;
+      await setupDatabase();
+
+      const rdsSchemaFilePath = path.join(projRoot, 'amplify', 'backend', 'api', apiName, 'schema.sql.graphql');
+
+      await addApiWithAllAuthModes(projRoot, { transformerVersion: 2, apiName });
+
+      await importRDSDatabase(projRoot, {
+        database,
+        engine,
+        host,
+        port,
+        username,
+        password,
+        useVpc: true,
+        apiExists: true,
+      });
+
+      const schema = /* GraphQL */ `
+        input AMPLIFY {
+          engine: String = "${engineName}"
+        }
+
+        # IAM Public can't delete a record
+        # IAM Public can't update SSN field
+        # IAM Public can't read SSN field 
+        # IAM Public can create a record with all fields, but can't read SSN response
+        # IAM Public subscription can't read SSN field
+        type Person1 @model @auth(rules: [{allow: private, provider: iam}, {allow: public, provider: iam}]) {
```

Comment:
For all Person* type tests: it looks like we're not testing that the private client has full access while the public client is appropriately restricted. Is the private client behavior covered elsewhere? We've seen odd behaviors with combinations in the past.

Reply:
Though it is not explicitly mentioned, the test also covers 'get', 'list' and 'create' operation on IAM private client. I will add test cases for 'update' and 'delete' for IAM::private.